jwst/saturation/saturation.py

```diff
@@ -160,6 +165,24 @@ def irs2_flag_saturation(input_model, ref_model, n_pix_grow_sat):
                                         irs2_mask, detector)
             # check for saturation
             flag_temp = np.where(sci_temp >= sat_thresh, SATURATED, 0)
+            if group == 2:
+                # Identify groups which we wouldn't expect to saturate by the third group,
+                # on the basis of the first group
+                mask = sci_temp[ints, 0, ...] / np.mean(read_pattern[0]) * read_pattern[2][-1] < sat_thresh
```

Comment:
I don't think this will work; sci_temp as defined above is 2048x2048, so accessing sci_temp[ints,0,...] will just return a single element [0,0] of the array.